### PR TITLE
rusty-psn, rusty-psn-gui: init at 0.1.2

### PR DESCRIPTION
**Pull request by AngryAnt**
###### Description of changes

A simple tool to grab updates for PS3 games, directly from Sony's servers using their updates API.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    ];
+  } else "";
+  desktopItems = if gui then [ desktopItem ] else [];
+
+  meta = with lib; {
+    description = "A simple tool to grab updates for PS3 games, directly from Sony's servers using their updates API";
+    homepage = "";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with lib.maintainers; [ AngryAnt ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:78]:
```suggestion
    maintainers = with maintainers; [ AngryAnt ];
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      "ps3"
+      "sony"
+      "playstation"
+      "update"
+    ];
+  } else "";
+  desktopItems = if gui then [ desktopItem ] else [];
+
+  meta = with lib; {
+    description = "A simple tool to grab updates for PS3 games, directly from Sony's servers using their updates API";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:74]:
```suggestion
    description = "Simple tool to grab updates for PS3 games, directly from Sony's servers using their updates API";
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    ];
+    keywords = [
+      "psn"
+      "ps3"
+      "sony"
+      "playstation"
+      "update"
+    ];
+  } else "";
+  desktopItems = if gui then [ desktopItem ] else [];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:71]:
```suggestion
  desktopItems = lib.optionals withGui [ desktopItem ];
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [ (if gui then "egui" else "cli") ];
+
+  postFixup = ''
+    patchelf --set-rpath "${lib.makeLibraryPath dependencies}" $out/bin/rusty-psn
+    ${if gui then "mv $out/bin/rusty-psn $out/bin/rusty-psn-gui" else ""}
+  '';
+
+  desktopItem = if gui then makeDesktopItem {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:55]:
```suggestion
  desktopItem = lib.optionalString makeDesktopItem {
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      "Network"
+    ];
+    keywords = [
+      "psn"
+      "ps3"
+      "sony"
+      "playstation"
+      "update"
+    ];
+  } else "";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:70]:
```suggestion
  };
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  ];
+  buildInputs = dependencies;
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [ (if gui then "egui" else "cli") ];
+
+  postFixup = ''
+    patchelf --set-rpath "${lib.makeLibraryPath dependencies}" $out/bin/rusty-psn
+    ${if gui then "mv $out/bin/rusty-psn $out/bin/rusty-psn-gui" else ""}
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:53]:
```suggestion
  '' + lib.optionalString withGui ''
    mv $out/bin/rusty-psn $out/bin/rusty-psn-gui
  '';
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    xorg.libxcb
+    libGL
+    libGL.dev
+  ] else [
+    openssl
+  ];
+  buildInputs = dependencies;
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [ (if gui then "egui" else "cli") ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:48]:
```suggestion
  buildFeatures = [ (if withGui then "egui" else "cli") ];
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+  postFixup = ''
+    patchelf --set-rpath "${lib.makeLibraryPath dependencies}" $out/bin/rusty-psn
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:51]:
```suggestion
    patchelf --set-rpath "${lib.makeLibraryPath buildInputs}" $out/bin/rusty-psn
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libxcb
+    libGL
+    libGL.dev
+  ] else [
+    openssl
+  ];
+  buildInputs = dependencies;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:45]:
```suggestion
  buildInputs = if gui then [
    openssl
    xorg.libxcb
    xorg.libX11
    xorg.libXcursor
    xorg.libXrandr
    xorg.libXi
    xorg.libxcb
    libGL
    libGL.dev
  ] else [
    openssl
  ];
```

File: pkgs/applications/misc/rusty-psn/default.nix [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, makeDesktopItem
+, copyDesktopItems
+, pkg-config
+, openssl
+, xorg
+, libGL
+, gui ? false # build GUI version
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rusty-psn/default.nix:11]:
```suggestion
, withGui ? false # build GUI version
```

**Comment by alyssais** [comment]
@OfBorg eval

**Comment by AngryAnt** [comment]
@ofborg build rusty-psn

**Pull request merged**